text/0000-dynamic-import-hints.md

```diff
@@ -0,0 +1,255 @@
+---
+title: Dynamic Import Hints and Its Metadata
+status: DRAFTED
+created_at: 2020-03-16
+updated_at: 2020-03-22
+rfc: https://github.com/salesforce/lwc-rfcs/pull/27
+champion: Aliaksandr Papko (@apapko) | LWC Team
+implementation: https://git.soma.salesforce.com/lwc/lwc-api/pull/42 & https://git.soma.salesforce.com/lwc/lwc-platform/pull/361
+---
+
+# Hints Syntax and Its Metadata
+
+Table of Content:
+* [Summary](#summary)
+* [Motivation](#motivation)
+* [Detailed Design](#detailed-design)
+    * [Grammar](#grammar)
+        * [Hint Syntax](#hint-syntax)
+        * [Static Semantics](#static-semantics)
+        * [Semantics](#semantics)
+        * [Examples](#examples)
+        * [Separation of Concern](#separation-of-concern)
+    * [Reasons](#reasons)
+    * [Dynamic Import Metadata](#dynamic-import-metadata)
+* [Alternate Syntax Considered](#alternate-syntax-considered)
+* [Future work](#future-work)
+* [References](#references)
+
+## Summary
+This RFC suggests authoring syntax for specifying dynamic import hints and describes its metadata shape.
+
+## Motivation
+Using hints allows component authors to specify a condition on which a dynamic import will be necessary. This is a way to describe the conditions required for this import to happen. An application framework can use these hints to optimize the loading of those resources (e.g.: pre-fetch at the time of resolving dependencies)
+
+## Hint Invariants
+* Hint key and value are string literals
+* Hints do not influence the outcome of the compilation phase. It only produces additional metadata for the application framework to consume.
+
+## Detailed Design
+
+### Grammar
+This section defines the grammar to be followed for expressing hints. The grammar establishes a contract between the component and the compiler to facilitate hint recognition by the compiler, gathering the metadata and providing that as part of the compilation result.
+
+#### Hint Syntax
```

Comment:
Consider defining syntax and grammars within RFCs using [ABNF](https://tools.ietf.org/html/rfc5234) since it's the standard IETF language for defining grammar and syntax rules.

Reply:
@jbenallen Thanks for the suggestion. I will merge this PR and use a follow up PR to update the grammar pieces in the ABNF format. This will isolate the styling changes.